jax/_src/pjit.py

```diff
@@ -807,8 +807,10 @@ def ax_leaf(l):
             isinstance(l, tuple) and all_leaves(l, lambda x: x is None))
   return broadcast_prefix(abstracted_axes, args, ax_leaf)
 
+_P = ParamSpec("_P")
```

Comment:
reuse this from stages instead of redefining in 3 files.

Reply:
Thanks! Addressed in 32de804c5c320942ef4b93d1f18751391edc2b27 (rebased to keep everything in a single commit)